avm/res/api-management/service/main.bicep

```diff
@@ -75,14 +75,8 @@ param roleAssignments roleAssignmentType
 ])
 param sku string = 'Premium'
 
-@description('Optional. The instance size of this API Management service. Not supported with V2 SKUs. If using Consumption, sku should = 0.')
-@allowed([
-  0
-  1
-  2
-  3
-])
-param skuCount int = 2
+@description('Conditional. The scale units for this API Management service. Required if using Basic, Standard, or Premium skus. For range of capacities for each sku, reference https://azure.microsoft.com/en-us/pricing/details/api-management/.')
```

Comment:
This is not defined clearly and under discussion even within the team, so not willing to block the PR on it, but just wanted to share my view.

If we add a default value to an input parameter, shouldn't the parameter type be Optional? At the end of the day, if the user does not provide any value, the default one is set. I see a Conditional type as a parameter for which a value is required to be passed only if a specific condition is satisfied.
In this case we are saying that the value is only set for sku Basic Standard and Premium. Developer and Consumer will be enforced to specific values.
If it was me I would add the above info to the description, keep the default value but make the parameter Optional.

Cc @AlexanderSehr as we had a brief discussion on this topic today

Reply:
Jap, we need to discuss thisd a bit further. One could also argue a conditional parameter is just required if something else is, regardless of it's value. You just know that in one case you **have** to set it, while in others you can or don't. Stay tuned :)

Reply:
Going to resolve since this is still a discussion. I'll leave it as-is for now.